autosklearn/automl.py

```diff
@@ -1343,6 +1343,66 @@ def score(self, X, y):
                                 task_type=self._task,
                                 metric=self._metric, )
 
+    def _get_runhistory_models_performance(self):
+        metric = self._metric
+        data = self.runhistory_.data
+        performance_list = []
+        for run_key, run_value in data.items():
+            if run_value.status != StatusType.SUCCESS:
+                # Ignore crashed runs
+                continue
+            # Alternatively, it is possible to also obtain the start time with ``run_value.starttime``
+            endtime = pd.Timestamp(time.strftime('%Y-%m-%d %H:%M:%S',
+                                                time.localtime(run_value.endtime)))
+            val_score = metric._optimum - (metric._sign * run_value.cost)
+            train_score = metric._optimum - (metric._sign * run_value.additional_info['train_loss'])
+            performance_list.append({
+                'Timestamp': endtime,
+                'single_best_optimization_score': val_score,
+                'single_best_train_score': train_score,
+            })
+            #append test-scores, if data for test_loss are available. This is the case, if X_test and y_test where provided.
+            if 'test_loss' in run_value.additional_info:
+                test_score = metric._optimum - (metric._sign * run_value.additional_info['test_loss'])
```

Comment:
Probably we can be robust against not having `test_loss` in the additional_info dictionary.

What I mean, is that we can do something like:
```
test_loss = np.nan
if test_loss' in run_value.additional_info:
    est_loss = <further code here>
```

Because, `X_test` might not be passed to Auto-sklearn

Reply:
I suppose you mean 

```
test_score = np.nan
if 'test_loss' in run_value.additional_info:
    test_score = <further code here>
```

Otherwise I dont know what you mean. My implementation is already robust against not having the key `'test_loss'`, which is the case when `X_test` is not passed. That is why I am checking for `'test_loss' in run_value.additional_info`